server/app/services/scheduler/strategy/random.rb

```diff
@@ -13,22 +13,32 @@ def host_grace_period
         30.seconds
       end
 
-      ##
       # @param [GridService] grid_service
       # @param [Integer] instance_number
-      # @param [Array<HostNode>] nodes
+      # @param [Array<Scheduler::Node>] nodes
+      # @return [Scheduler::Node, NilClass]
       def find_node(grid_service, instance_number, nodes)
         if grid_service.stateless?
+          node = find_previous_or_new_node(grid_service, instance_number, nodes)
+          return node if node
```

Comment:
Comparing this to existing Kontena 1.1 behavior, it would just ignore the affinity filter in this case, and keep scheduling the stateful service onto the existing node. So I can't quite tell if this is any more or less broken for the slightly crazy case of a stateful random service with changing affinity filters.